packages/web/app/environment.ts

```diff
@@ -0,0 +1,164 @@
+import zod from 'zod';
+
+const BaseSchema = zod.object({
+  NODE_ENV: zod.string(),
+  ENVIRONMENT: zod.string(),
+  APP_BASE_URL: zod.string().url(),
+  GRAPHQL_ENDPOINT: zod.string().url(),
+  EMAILS_ENDPOINT: zod.string().url(),
+  SUPERTOKENS_CONNECTION_URI: zod.string().url(),
+  SUPERTOKENS_API_KEY: zod.string(),
+  SLACK_CLIENT_ID: zod.string(),
+  SLACK_CLIENT_SECRET: zod.string(),
+  GITHUB_APP_NAME: zod.string(),
+  GA_TRACKING_ID: zod.string().optional(),
+  CRISP_WEBSITE_ID: zod.string().optional(),
+  DOCS_URL: zod.string().url().optional(),
+  STRIPE_PUBLIC_KEY: zod.string().optional(),
+  RELEASE: zod.string().optional(),
+});
+
+const AuthGitHubConfigSchema = zod.union([
+  zod.object({
+    AUTH_GITHUB: zod.union([zod.void(), zod.literal('0')]),
+  }),
+  zod.object({
+    AUTH_GITHUB: zod.literal('1'),
+    AUTH_GITHUB_CLIENT_ID: zod.string(),
+    AUTH_GITHUB_CLIENT_SECRET: zod.string(),
+  }),
+]);
+
+const AuthGoogleConfigSchema = zod.union([
+  zod.object({
+    AUTH_GOOGLE: zod.union([zod.void(), zod.literal('0')]),
+  }),
+  zod.object({
+    AUTH_GOOGLE: zod.literal('1'),
+    AUTH_GOOGLE_CLIENT_ID: zod.string(),
+    AUTH_GOOGLE_CLIENT_SECRET: zod.string(),
+  }),
+]);
+
+const SentryConfigSchema = zod.union([
+  zod.object({
+    SENTRY: zod.union([zod.void(), zod.literal('0')]),
+  }),
+  zod.object({
+    SENTRY: zod.literal('1'),
+    SENTRY_DSN: zod.string(),
+  }),
+]);
+
+const LegacyAuth0Config = zod.union([
+  zod.object({
+    AUTH_LEGACY_AUTH0: zod.union([zod.void(), zod.literal('0')]),
+  }),
+  zod.object({
+    AUTH_LEGACY_AUTH0: zod.literal('1'),
+    AUTH_LEGACY_AUTH0_AUDIENCE: zod.string(),
+    AUTH_LEGACY_AUTH0_ISSUER_BASE_URL: zod.string(),
+    AUTH_LEGACY_AUTH0_CLIENT_ID: zod.string(),
+    AUTH_LEGACY_AUTH0_CLIENT_SECRET: zod.string(),
+    AUTH_LEGACY_AUTH0_INTERNAL_API_ENDPOINT: zod.string(),
+    AUTH_LEGACY_AUTH0_INTERNAL_API_KEY: zod.string(),
+  }),
+]);
+
+const configs = {
+  base: BaseSchema.safeParse(process.env),
+  sentry: SentryConfigSchema.safeParse(process.env),
+  authGithub: AuthGitHubConfigSchema.safeParse(process.env),
+  authGoogle: AuthGoogleConfigSchema.safeParse(process.env),
+  authLegacyAuth0: LegacyAuth0Config.safeParse(process.env),
+};
+
+const environmentErrors: Array<string> = [];
+
+for (const config of Object.values(configs)) {
+  if (config.success === false) {
+    environmentErrors.push(JSON.stringify(config.error.format(), null, 4));
```

Comment:
We should figure out better formatting here :)

Reply:
It should be okay for now - but in the future, we definitely should make this more obvious.